engines/dfc_provider/spec/requests/orders_backorder_spec.rb

```diff
@@ -0,0 +1,112 @@
+# frozen_string_literal: true
+
+require_relative "../spec_helper"
+
+RSpec.describe "Orders integration" do
+  let(:user) { create(:oidc_user, id: 12_345) }
+  let(:supplier) {
+    create(
+      :distributor_enterprise, # supplier sells their products on an OFN instance
+      id: 10_000, owner: user, name: "Fred's Farm",
+      address: build(:address, id: 40_000),
+    )
+  }
+  let(:product) {
+    create(
+      :base_product,
+      id: 90_000, name: "Apple", description: "Red",
+      variants: [variant],
+    )
+  }
+  let(:variant) { build(:base_variant, id: 10_001, unit_value: 1, sku: "AR", supplier:) }
+
+  let(:distributor) {
+    create(
+      :distributor_enterprise, # distributor has imported supplier's product to create a copy of it
+      id: 11_000, owner: user, name: "Shane's Shop",
+      address: build(:address, id: 41_000),
+    )
+  }
+  let(:distributor_product) {
+    create(
+      :base_product,
+      id: 91_000, name: "Apple", description: "Red",
+      variants: [variant],
+    )
+  }
+  let(:distributor_variant) { build(:base_variant, id: 11_001, unit_value: 1, sku: "AR", supplier:) }
+
+  before {
+    login_as user
+    product
+
+    # TODO: create distributor product with semantic link to supplier product
+  }
+
+  describe BackorderJob do
+    it "creates a product" do
+      # post(enterprise_orders_path(supplier.id))
+      pending "finish spec"
+      # TODO: make a distributor order, and flush BackorderJob
+
+      # A backorder to the supplier has been created
+      expect(supplier.distributed_orders.count).to eq 1
+    end
+  end
+end
+
+
+RSpec.describe "ORders FdcBackorderer", type: :request do
+  host = Rails.application.default_url_options[:host]
+
+  let(:backorderer) { FdcBackorderer.new(order.distributor.owner, urls) }
+  let(:urls) { FdcUrlBuilder.new(product_link) }
+  let(:product_link) {
+    "http://#{host}/api/dfc/enterprises/#{supplier.id}/catalog_items" #why is not acessible? normally request specs are on port 80
+  }
```

Comment:
From what I'm reading, request specs don't run a server like puma and therefore we don't have a host and port, we don't have Capybara and no HTTP interaction.

So if we want to test with our BackorderJob or anything else that uses the DfcRequest object then we need to use a system spec. Only then we can use HTTP.

The next challenge is that our app uses `test.host` as domain but it's not accessible under that domain. If we want a proper integration test, we would need to set the right IP address and port.